app/domain/authentication/resource_restrictions/extract_resource_restrictions.rb

```diff
@@ -26,6 +27,7 @@ def call
         fetch_resource_annotations
         extract_resource_restrictions_from_annotations
         create_resource_restrictions_object
+        validate_restriction_configuration
```

Comment:
can you explain please why you are calling this method here instead of calling it just when it's needed in the calling class?